### PR TITLE
feat: Package init segments separately

### PR DESCRIPTION
**Pull request by lfordyce**

### CODE REVIEW

**Comment by jensenn** [comment]
I assume most of the code in mp4_init_muxer.cc/.h is a subset copied from elsewhere in the library. Is there anything major that was changed in that code?

File: packager/media/formats/mp4/mp4_muxer.h [hunk 1]
```diff
@@ -36,6 +36,23 @@ class MP4Muxer : public Muxer {
   explicit MP4Muxer(const MuxerOptions& options);
   ~MP4Muxer() override;
 
+ protected:
+  // Get time in seconds since midnight, Jan. 1, 1904, in UTC Time.
+  uint64_t IsoTimeNow();
```
Comment (chenda6) [inline packager/media/formats/mp4/mp4_muxer.h:41]: It doesn't look like we need to elevate these to `protected`.

File: packager/media/formats/mp4/mp4_init_muxer.cc [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+namespace shaka {
+namespace media {
+namespace mp4 {
+
+MP4InitMuxer::MP4InitMuxer(const MuxerOptions& options) : MP4Muxer(options) {}
+
+MP4InitMuxer::~MP4InitMuxer() {}
+
+Status MP4InitMuxer::Finalize() {
+  if (!segmenter_) {
```
Comment (chenda6) [inline packager/media/formats/mp4/mp4_init_muxer.cc:23]: It looks like neither `segmenter_` nor `to_be_initialized_` is needed by this class?  I think we can simplify this further by removing them here and reverting them in `MP4Muxer`.

**Comment by chenda6** [comment]
Good investigative work and a clean solution to support packaging init segments only.  
@jensenn @Vadym-Kupriyanchuk: would like hear your feedback.

**Comment by jensenn** [comment]
I like this approach much better. LGTM

**Review by jensenn — APPROVED** [review]

**Pull request merged**